android/capacitor/src/main/java/com/getcapacitor/plugin/SystemBars.java

```diff
@@ -0,0 +1,209 @@
+package com.getcapacitor.plugin;
+
+import android.content.pm.PackageInfo;
+import android.content.res.Configuration;
+import android.view.View;
+import android.view.Window;
+import androidx.core.graphics.Insets;
+import androidx.core.view.ViewCompat;
+import androidx.core.view.WindowCompat;
+import androidx.core.view.WindowInsetsCompat;
+import androidx.core.view.WindowInsetsControllerCompat;
+import androidx.webkit.WebViewCompat;
+import com.getcapacitor.Plugin;
+import com.getcapacitor.PluginCall;
+import com.getcapacitor.PluginMethod;
+import com.getcapacitor.annotation.CapacitorPlugin;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@CapacitorPlugin
+public class SystemBars extends Plugin {
+
+    static final String STYLE_LIGHT = "LIGHT";
+    static final String STYLE_DARK = "DARK";
+    static final String STYLE_DEFAULT = "DEFAULT";
+    static final String BAR_STATUS_BAR = "StatusBar";
+    static final String BAR_GESTURE_BAR = "NavigationBar";
+
+    static final String viewportMetaJSFunction = """
+        function capacitorSystemBarsCheckMetaViewport() {
+            const meta = document.querySelectorAll("meta[name=viewport]");
+            if (meta.length == 0) {
+                return false;
+            }
+            // get the last found meta viewport tag
+            const metaContent = meta[meta.length - 1].content;
+            return metaContent.includes("viewport-fit=cover");
+        }
+
+        capacitorSystemBarsCheckMetaViewport();
+        """;
+
+    @Override
+    public void load() {
+        super.load();
+        initSystemBars();
+    }
+
+    private boolean hasFixedWebView() {
+        PackageInfo packageInfo = WebViewCompat.getCurrentWebViewPackage(bridge.getContext());
+        Pattern pattern = Pattern.compile("(\\d+)");
+        Matcher matcher = pattern.matcher(packageInfo.versionName);
+
+        if (!matcher.find()) {
+            return false;
+        }
+
+        String majorVersionStr = matcher.group(0);
+        int majorVersion = Integer.parseInt(majorVersionStr);
+
+        return majorVersion >= 140;
+    }
+
+    private void initSystemBars() {
+        String style = getConfig().getString("style", STYLE_DEFAULT).toUpperCase(Locale.US);
+        boolean hidden = getConfig().getBoolean("hidden", false);
+        boolean disableCSSInsets = getConfig().getBoolean("disableInsets", false);
+
+        this.bridge.getWebView().evaluateJavascript(viewportMetaJSFunction, (res) -> {
+            boolean hasMetaViewportCover = res.equals("true");
+            if (!disableCSSInsets) {
+                setupSafeAreaInsets(this.hasFixedWebView(), hasMetaViewportCover);
+            }
+        });
+
+        getBridge().executeOnMainThread(() -> {
+            setStyle(style, "");
+            setHidden(hidden, "");
+        });
+    }
+
+    @PluginMethod
```

Comment:
The swift equivalents of the plugin methods here are annotated with a `returnType: none`. Would it make sense to change it to `@PluginMethod(returnType = PluginMethod.RETURN_NONE)`?